dspace-server-webapp/src/test/java/org/dspace/app/rest/csv/CSVMetadataImportReferenceTest.java

```diff
@@ -0,0 +1,328 @@
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.UUID;
+
+import org.dspace.app.rest.builder.CollectionBuilder;
+import org.dspace.app.rest.builder.CommunityBuilder;
+import org.dspace.app.rest.builder.ItemBuilder;
+import org.dspace.app.rest.test.AbstractEntityIntegrationTest;
+import org.dspace.content.Collection;
+import org.dspace.content.Item;
+import org.dspace.content.MetadataField;
+import org.dspace.content.MetadataValue;
+import org.dspace.content.Relationship;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.ItemService;
+import org.dspace.content.service.MetadataFieldService;
+import org.dspace.content.service.MetadataValueService;
+import org.dspace.content.service.RelationshipService;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Created by: Andrew Wood
+ * Date: 26 Jul 2019
+ */
+public class CSVMetadataImportReferenceTest extends AbstractEntityIntegrationTest {
+
+    //Common collection to utilize for test
+    private Collection col1;
+
+    @Autowired
+    private RelationshipService relationshipService;
+
+    @Autowired
+    private ItemService itemService;
+
+    /**
+     * Setup testing enviorment
+     */
+    @Before
+    public void setup() {
+        context.turnOffAuthorisationSystem();
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                .withName("Parent Community")
+                .build();
+        col1 = CollectionBuilder.createCollection(context, parentCommunity).withName("Collection 1").build();
+    }
+
+    /**
+     * Helper method to validate relationships
+     * @param leftItem the left item in a known relationship
+     * @param rightItem the right item in a known relationship
+     * @param expectedCount expected relationship count for a known relationship
+     */
+    private void assertRelationship(Item leftItem, Item rightItem, int expectedCount) throws SQLException {
+        List<Relationship> rels = relationshipService.findByItem(context, leftItem);
+        int foundCount = 0;
+        for (Relationship rel : rels) {
+            if (rel.getLeftItem().getID().equals(leftItem.getID())
+                    && rel.getRightItem().getID().equals(rightItem.getID())) {
+                foundCount++;
+            }
+        }
+        assertEquals(expectedCount, foundCount);
+    }
+
+    /**
+     * Test existence of newly created items with proper relationships defined in mets via
+     * a metadata field and value reference
+     */
+    @Test
+    public void testSingleMdRef() throws Exception {
+        String[] csv = {"id,relationship.type,relation.isAuthorOfPublication,collection,dc.identifier.other",
+                "+,Person,," +  col1.getHandle() + ",0",
+                "+,Publication,dc.identifier.other:0," +  col1.getHandle() + ",1"};
```

Comment:
can we also test this will fail correctly if the referenced person occurs in the CSV after the publication